src/app/core/json-patch/json-patch-operations.service.ts

```diff
@@ -0,0 +1,129 @@
+                  operationsList.children[resourceId].body.forEach((entry) => {
+                    body.push(entry.operation);
+                  });
+                }
+              } else {
+                Object.keys(operationsList.children)
+                  .filter((key) => operationsList.children.hasOwnProperty(key))
+                  .filter((key) => hasValue(operationsList.children[key]))
+                  .filter((key) => hasValue(operationsList.children[key].body))
+                  .forEach((key) => {
+                    operationsList.children[key].body.forEach((entry) => {
+                      body.push(entry.operation);
+                    });
+                  })
+              }
+            }
+            return this.getRequestInstance(this.requestService.generateRequestId(), endpointURL, body);
+          }));
+      })));
+
+    return observableMerge(
+      emptyRequest$.pipe(
+        filter((request: PatchRequestDefinition) => isEmpty(request.body)),
+        tap(() => startTransactionTime = null),
+        map(() => null)),
+      patchRequest$.pipe(
+        filter((request: PatchRequestDefinition) => isNotEmpty(request.body)),
+        tap(() => this.store.dispatch(new StartTransactionPatchOperationsAction(resourceType, resourceId, startTransactionTime))),
+        tap((request: PatchRequestDefinition) => this.requestService.configure(request, true)),
+        flatMap((request: PatchRequestDefinition) => {
+          const [successResponse$, errorResponse$] = partition((response: RestResponse) => response.isSuccessful)(this.responseCache.get(request.href).pipe(
+            filter((entry: ResponseCacheEntry) => startTransactionTime < entry.timeAdded),
+            take(1),
+            map((entry: ResponseCacheEntry) => entry.response)
+          ));
+          return observableMerge(
+            errorResponse$.pipe(
+              tap(() => this.store.dispatch(new RollbacktPatchOperationsAction(resourceType, resourceId))),
+              flatMap((response: ErrorResponse) => observableOf(new Error(`Couldn't patch operations`)))),
+            successResponse$.pipe(
+              filter((response: PostPatchSuccessResponse) => isNotEmpty(response)),
+              tap(() => this.store.dispatch(new CommitPatchOperationsAction(resourceType, resourceId))),
+              map((response: PostPatchSuccessResponse) => response.dataDefinition),
+              distinctUntilChanged()));
+        }))
+    );
+  }
+
+  protected getRequestInstance(uuid: string, href: string, body?: any): PatchRequestDefinition {
+    return new this.patchRequestConstructor(uuid, href, body);
+  }
+
+  protected getEndpointByIDHref(endpoint, resourceID): string {
+    return isNotEmpty(resourceID) ? `${endpoint}/${resourceID}` : `${endpoint}`;
+  }
+
+  public jsonPatchByResourceType(linkName: string, scopeId: string, resourceType: string): Observable<ResponseDefinitionDomain> {
```

Comment:
Please add TypeDoc/comments to all your public methods in this file.  Ideally, all methods should have a brief description, if possible.